update description (#1)
DevExpress-Examples/how-to-implement-cascaded-combo-box-columns-in-aspxgridview-without-using-templates-e3689: Readme.md

> # ASPxGridView for ASP.NET Web Forms - Cascading Combo Boxes

Reviewer: ```suggestion
# Grid View for ASP.NET Web Forms - Cascading Combo Boxes
```

<!-- run online -->
**[[Run Online]](https://codecentral.devexpress.com/e3689/)**
<!-- run online end -->

This example demonstrates how to implement cascading combo box editors within [ASPxGridView](https://docs.devexpress.com/AspNet/DevExpress.Web.ASPxGridView).

> ![CascadingComboBoxes](~/images/CascadingComboBoxes.png)

Reviewer: ```suggestion
![CascadingComboBoxes](images/CascadingComboBoxes.png)
```

## Setup the Grid and its Column Editors

> Create an [ASPxGridView](https://docs.devexpress.com/AspNet/DevExpress.Web.ASPxGridView) control, assign its data source, and set the grid's [edit mode](https://docs.devexpress.com/AspNet/DevExpress.Web.ASPxGridViewEditingSettings.Mode) to `Inline`. Add two columns of the [GridViewDataComboBoxColumn](https://docs.devexpress.com/AspNet/DevExpress.Web.GridViewDataComboBoxColumn) type and set their data sources.

Reviewer 2: ```suggestion
Create an [ASPxGridView](https://docs.devexpress.com/AspNet/DevExpress.Web.ASPxGridView) control, assign its data source, and set the grid's [edit mode](https://docs.devexpress.com/AspNet/DevExpress.Web.ASPxGridViewEditingSettings.Mode) to `Inline`. Add two columns of the [GridViewDataComboBoxColumn](https://docs.devexpress.com/AspNet/DevExpress.Web.GridViewDataComboBoxColumn) type and specify their data sources.
```

```xml
<dx:ASPxGridView ID="Grid" runat="server" AutoGenerateColumns="false" DataSourceID="Customers"
    KeyFieldName="CustomerID" ClientInstanceName="grid">
    <Columns>
        <%--...--%>
        <dx:GridViewDataComboBoxColumn Caption="Country" FieldName="CountryID">
            <PropertiesComboBox DataSourceID="Countries" ValueField="CountryID" ValueType="System.Int32"
                <%--...--%>
            </PropertiesComboBox>
        </dx:GridViewDataComboBoxColumn>
        <dx:GridViewDataComboBoxColumn Caption="City" FieldName="CityID">
            <PropertiesComboBox DataSourceID="AllCities" ValueField="CityID" ValueType="System.Int32"
                <%--...--%>
            </PropertiesComboBox>
        </dx:GridViewDataComboBoxColumn>
    </Columns>
    <SettingsEditing Mode="Inline" />
</dx:ASPxGridView>
<asp:ObjectDataSource ID="Customers" runat="server" ... />
<asp:ObjectDataSource ID="Countries" runat="server" ... />
<asp:ObjectDataSource ID="AllCities" runat="server" ... />
```

## Respond to a Selection Change on the Client Side

> Handle the primary editor's client-side [SelectedIndexChanged](https://docs.devexpress.com/AspNet/js-ASPxClientComboBox.SelectedIndexChanged) event. In this event handler, get the current editor's value (the [GetValue](https://docs.devexpress.com/AspNet/js-ASPxClientEditBase.GetValue) method) and pass it as a parameter in the [PerformCallback](https://docs.devexpress.com/AspNet/js-ASPxClientComboBox.PerformCallback(parameter)) method of the secondary editor. To access the secondary editor, call the [GetEditor](https://docs.devexpress.com/AspNet/js-ASPxClientGridView.GetEditor(column)) method.

Reviewer: current editor's value - its a value of the current editor. I think, you mean current value of editor. Anyway, I would remove the `current` at all
```suggestion
Handle the primary editor's client-side [SelectedIndexChanged](https://docs.devexpress.com/AspNet/js-ASPxClientComboBox.SelectedIndexChanged) event. In this event handler, get the editor value (the [GetValue](https://docs.devexpress.com/AspNet/js-ASPxClientEditBase.GetValue) method) and pass it as a parameter in the [PerformCallback](https://docs.devexpress.com/AspNet/js-ASPxClientComboBox.PerformCallback(parameter)) method of the secondary editor. To access the secondary editor, call the [GetEditor](https://docs.devexpress.com/AspNet/js-ASPxClientGridView.GetEditor(column)) method.
```

```xml
<dx:ASPxGridView ID="Grid" runat="server" ... >
    <Columns>
        <%--...--%>
        <dx:GridViewDataComboBoxColumn Caption="Country" FieldName="CountryID" ...>
            <PropertiesComboBox DataSourceID="Countries" ValueField="CountryID" ...>
                <ClientSideEvents SelectedIndexChanged="CountriesCombo_SelectedIndexChanged" />
            </PropertiesComboBox>
        </dx:GridViewDataComboBoxColumn>
        <dx:GridViewDataComboBoxColumn Caption="City" FieldName="CityID" ... />
    </Columns>
</dx.ASPxGridView>
```

```js
function CountriesCombo_SelectedIndexChanged(s, e) {
    grid.GetEditor("CityID").PerformCallback(s.GetValue());
}
```

## Filter the Secondary Combo Box Values on the Server Side

> In the grid's [CellEditorInitialize](https://docs.devexpress.com/AspNet/DevExpress.Web.ASPxGridView.CellEditorInitialize) event handler, access the secondary editor and handle its [Callback](https://docs.devexpress.com/AspNet/DevExpress.Web.ASPxCallback.Callback) event. In this event handler, use the [Parameter](https://docs.devexpress.com/AspNet/DevExpress.Web.CallbackEventArgsBase.Parameter) argument property to obtain the primary editor's value from the client side. Filter the secondary editor's data source based on this value.

Reviewer: ```suggestion
In the [CellEditorInitialize](https://docs.devexpress.com/AspNet/DevExpress.Web.ASPxGridView.CellEditorInitialize) event handler, access the secondary editor and add a handler to its [Callback](https://docs.devexpress.com/AspNet/DevExpress.Web.ASPxCallback.Callback) event. In the handler, use the [Parameter](https://docs.devexpress.com/AspNet/DevExpress.Web.CallbackEventArgsBase.Parameter) argument property to obtain the primary editor's value from the client side. Filter the secondary editor's data source based on this value and bind the filtered values to the editor.
```
